modin/pandas/test/dataframe/test_map_metadata.py

```diff
@@ -557,6 +558,57 @@ def test_astype_category_large():
     assert modin_result.dtypes.equals(pandas_result.dtypes)
 
 
+@pytest.mark.parametrize("infer_objects", (True, False), ids=("infer_objects", ""))
+@pytest.mark.parametrize("convert_string", (True, False), ids=("convert_string", ""))
+@pytest.mark.parametrize("convert_integer", (True, False), ids=("convert_integer", ""))
+@pytest.mark.parametrize("convert_boolean", (True, False), ids=("convert_boolean", ""))
+@pytest.mark.parametrize(
+    "convert_floating", (True, False), ids=("convert_floating", "")
+)
+def test_convert_dtypes(
+    infer_objects, convert_string, convert_integer, convert_boolean, convert_floating
+):
+    # Sanity check, copied from pandas documentation:
+    # https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.convert_dtypes.html
+    data = {
+        "a": pd.Series([1, 2, 3], dtype=np.dtype("int32")),
+        "b": pd.Series(["x", "y", "z"], dtype=np.dtype("O")),
+        "c": pd.Series([True, False, np.nan], dtype=np.dtype("O")),
+        "d": pd.Series(["h", "i", np.nan], dtype=np.dtype("O")),
+        "e": pd.Series([10, np.nan, 20], dtype=np.dtype("float")),
+        "f": pd.Series([np.nan, 100.5, 200], dtype=np.dtype("float")),
+    }
+    kwargs = {
+        "infer_objects": infer_objects,
+        "convert_string": convert_string,
+        "convert_integer": convert_integer,
+        "convert_boolean": convert_boolean,
+        "convert_floating": convert_floating,
+    }
+    modin_df = pd.DataFrame(data)
+    pandas_df = pandas.DataFrame(data)
+    modin_result = modin_df.convert_dtypes(**kwargs)
+    pandas_result = pandas_df.convert_dtypes(**kwargs)
+    assert modin_result.dtypes.equals(pandas_result.dtypes)
+
+
+def test_convert_dtypes_row_partition():
+    # Tests `convert_dtypes` when values in the same column may come from different
+    # row partitions.
+    part_col_size = MinPartitionSize.get()
+    # Column 0 should have string dtype
+    modin_part1 = pd.DataFrame(["a"] * part_col_size).convert_dtypes()
+    # Column 0 should have an int dtype
+    modin_part2 = pd.DataFrame([1] * part_col_size).convert_dtypes()
+    modin_result = pd.concat([modin_part1, modin_part2]).convert_dtypes()
+    pandas_part1 = pandas.DataFrame(["a"] * part_col_size).convert_dtypes()
+    pandas_part2 = pandas.DataFrame([1] * part_col_size).convert_dtypes()
+    pandas_result = pandas.concat([pandas_part1, pandas_part2]).convert_dtypes()
```

Comment:
```suggestion
    modin_df = pd.concat([modin_part1, modin_part2])
    pandas_df = pandas.DataFrame(["a", 1])
    # The initial dataframes should be the same
    df_equals(modin_df, pandas_df)
    # TODO(https://github.com/modin-project/modin/pull/3805): delete
    # this assert once df_equals checks dtypes    
    assert modin_df.dtypes.equals(pandas_df.dtypes)
    modin_result = modin_df.convert_dtypes()
    pandas_result = pandas_df.convert_dtypes()
    df_equals(modin_result, pandas_result)
    assert modin_result.dtypes.equals(pandas_result.dtypes)    
```

Reply:
It seems that the pandas_df also has to be constructed with `pandas.concat`, or else the index types don't match: the one constructed by concat uses `Int64Index([0, 0], dtype='int64')`, whereas the one constructed directly from a 2-element list has `RangeIndex(start=0, stop=2, step=1)`.

Reply:
Let's just do `pandas_df = pandas.DataFrame(["a", 1], index=[0, 0])` then.